### PR TITLE
depcheck action workflow

### LINKED ISSUE #869: depcheck github action
**Problem:** We have instances where we install a frontend library with `yarn`, but then we don't actually end up using it in our code. We should try to keep our frontend package size as small as possible and keep only the essentials to make our builds run faster

**Solution:** Use depcheck to create a Github Action that validates that there are no frontend libraries in `yarn.lock` that really don't get used. 

**Branch Setup:**
```
git checkout nextjs
git pull origin nextjs
git checkout -b depcheck-action
```

### PR DESCRIPTION
**Pull request by prishavall**
Github Issue: #869

We have instances where we install a frontend library with yarn, but then we don't actually end up using it in our code. We should try to keep our frontend package size as small as possible and keep only the essentials to make our builds run faster

It provides a new command npx dep-check to see if there are any unused dependencies

Testing Methodology
Ran it through the front-end directory

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "depcheck": "^1.4.7"
```
Comment (karkir0003) [inline package.json]: we should install depcheck under frontend/package.json

**Review by karkir0003 — COMMENTED** [review]
left a comment

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -9,8 +9,8 @@ on:
       - "frontend/**"
   pull_request:
     paths:
-      - "frontend/**"
-
+       - "frontend/**"
+       - ".github/workflows/node.js.yml"
```
Comment (farisdurrani) [inline .github/workflows/node.js.yml]: I dont see why it’s necessary to rerun the workflow on this very specific file. You probably only left this for testing. Recommend to remove

**Review by farisdurrani — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (karkir0003) [reply .github/workflows/node.js.yml]: this is for if we modify the yml file itself @farisdurrani

Comment (karkir0003) [reply .github/workflows/node.js.yml]: but if we need to remove it, we can

**Comment by karkir0003** [comment]
@farisdurrani are the package.json files correctly updated? @prishavall needed to add depcheck as a dev dependency?

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (farisdurrani) [reply .github/workflows/node.js.yml]: I just don't want to make a precedent all github workflow files are triggered if the specific workflow file is changed. It's an odd convention

**Comment by farisdurrani** [comment]
Yes, it's more apt to put in devDep since it's not a critical functionality for the app

File: .github/workflows/node.js.yml (hunk 2 above)
Comment (karkir0003) [reply .github/workflows/node.js.yml]: ok. so we should remove that line @farisdurrani

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -47,5 +47,9 @@ jobs:
       - name: Lint checks
         run: pnpm lint
         working-directory: ./frontend
+
+      - name: Run depcheck
+        run: npx depcheck --ignore-bin-package
+        working-directory: ./frontend
         
       # future: add yarn build once build errors are resolved
```
Comment (karkir0003) [inline .github/workflows/node.js.yml:54]: do we need to worry abt this todo @dwu359 @farisdurrani

**Review by karkir0003 — COMMENTED** [review]
@dwu359 @farisdurrani added another comment

File: .github/workflows/node.js.yml (hunk 3 above)
Comment (farisdurrani) [reply .github/workflows/node.js.yml:54]: Yes, in prod we would usually publish the built package obtained from `pnpm next build` but that command is failing. We have to fix that before we can implement the todo

Comment (karkir0003) [reply .github/workflows/node.js.yml:54]: but for this immediate action item, we don't need to worry about this for the scope of this PR? @farisdurrani

Comment (farisdurrani) [reply .github/workflows/node.js.yml:54]: You are correct

**Review by karkir0003 — APPROVED** [review]
LGTM! Good job on this :)

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
Lgtm

**Review by dwu359 — APPROVED** [review]

**Pull request merged**